src/main/java/com/dtolabs/rundeck/plugin/resources/ec2/InstanceToNodeMapper.java

```diff
@@ -104,42 +104,50 @@ public NodeSetImpl performQuery() {
             }
         }
 
-        if (getEndpoint().equals("ALL_REGIONS")) {
+        if(getEndpoint() != null) {
+            if (getEndpoint().equals("ALL_REGIONS")) {
 
-            //Retrieve dynamic list of EC2 regions from AWS
-            DescribeRegionsResult regionsResult = ec2.describeRegions();
-            for (Region region : regionsResult.getRegions()) {
-                regions.add(region.getEndpoint());
-            }
+                //Retrieve dynamic list of EC2 regions from AWS
+                DescribeRegionsResult regionsResult = ec2.describeRegions();
+                for (Region region : regionsResult.getRegions()) {
+                    regions.add(region.getEndpoint());
+                }
 
-        } else {
-            try {
-                //Use comma-separated list of region supplied by user
-                regions.addAll(Arrays.asList(getEndpoint().replaceAll("\\s+", "").split(",")));
-            } catch (NullPointerException e) {
-                throw new IllegalArgumentException("Failed to parse endpoint: Region cannot be empty");
+            } else {
+                try {
+                    //Use comma-separated list of region supplied by user
+                    regions.addAll(Arrays.asList(getEndpoint().replaceAll("\\s+", "").split(",")));
+                } catch (NullPointerException e) {
```

Comment:
If I am understanding what Forrest wrote in the Slack thread correctly, we actually want to change this to populate a default region, right? Because that was the previous functionality? Because at this point in the if statement, this case can essentially only be it trying to parse an empty field